File: app/src/main/java/com/mona/moviesapp/popular_people/FullImageActivity.java
package com.mona.moviesapp.popular_people;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.ImageView;

import com.mona.moviesapp.R;

import java.io.IOException;
import java.io.InputStream;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.concurrent.ExecutionException;

public class FullImageActivity extends AppCompatActivity {

    ImageView full_img;
    String picturePath;

    URL ImageUrl = null;
    InputStream inputStream = null;
    Bitmap bmImg = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_full_image);

        full_img = findViewById(R.id.fullimg);

        Intent intent = getIntent();
        Bundle bundle =  intent.getBundleExtra("data");
        if(!bundle.isEmpty()) {
            picturePath = bundle.getString("picture_path");
        }

        try {
            full_img.setImageBitmap(new downloadImage(full_img).execute(picturePath).get());
        } catch (ExecutionException e) {
            e.printStackTrace();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        try {
            new downloadImage(full_img).execute(picturePath).get();
        } catch (ExecutionException e) {
            e.printStackTrace();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

    }

    public class downloadImage extends AsyncTask<String, Void, Bitmap> {

        public downloadImage(ImageView imageView) {

        }

        @Override
        protected Bitmap doInBackground(String... strings) {

            try {
                ImageUrl = new URL(strings[0]);
                Log.i("URL",ImageUrl.toString());

                HttpURLConnection conn = (HttpURLConnection) ImageUrl.openConnection();
                conn.setDoInput(true);
                conn.connect();
                inputStream = conn.getInputStream();
                bmImg = BitmapFactory.decodeStream(inputStream);

            } catch (MalformedURLException e) {
                e.printStackTrace();
            } catch (IOException e) {
                e.printStackTrace();
            }
            return bmImg;
        }

        @Override
        protected void onPostExecute(Bitmap bitmap) {
            super.onPostExecute(bitmap);
            if(bitmap != null){
                full_img.setImageBitmap(bitmap);
            }
            else{
                full_img.setImageResource(R.drawable.ic_launcher_background);
            }
        }
    }
}

File: app/src/main/java/com/mona/movies_app/popular_people/full_image_screen/view/FullImageActivity.kt
package com.mona.movies_app.popular_people.full_image_screen.view

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.provider.MediaStore
import android.support.v4.app.ActivityCompat
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.ImageView
import android.widget.Toast
import com.bumptech.glide.Glide
import com.mona.movies_app.R
import com.mona.movies_app.popular_people.full_image_screen.interfaces.FullImageViewInterface
import com.mona.movies_app.popular_people.full_image_screen.model.FullImageModel
import com.mona.movies_app.popular_people.full_image_screen.presenter.FullImagePresenter

class FullImageActivity : AppCompatActivity(), FullImageViewInterface {

    private lateinit var fullImg: ImageView
    private lateinit var saveImg: Button

    private var detailsIntent: Intent? = null
    private var bundle: Bundle? = null

    private var picturePath: String? = ""

    private lateinit var fullImagePresenter: FullImagePresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_full_image)

        fullImg = findViewById(R.id.full_img)
        saveImg = findViewById(R.id.btn_save)

        fullImagePresenter = FullImagePresenter(this, FullImageModel())

        detailsIntent = intent
        bundle = intent!!.getBundleExtra("data")

        if (!bundle!!.isEmpty) {
            picturePath = bundle!!.getString("picture_path")
        }

        Glide.with(this)
                .load(picturePath)
                .into(fullImg)

        saveImg.setOnClickListener {
            if (ContextCompat.checkSelfPermission(this@FullImageActivity, Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED) {
            } else {
                ActivityCompat.requestPermissions(this@FullImageActivity, arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE), 0)
            }
//            fullImagePresenter.saveImage(picturePath!!)
            saveImageToGallery()
        }
    }

    override fun saveImageToGallery() {
        fullImg.isDrawingCacheEnabled = true
        val b = fullImg.drawingCache
        MediaStore.Images.Media.insertImage(contentResolver, b, picturePath, "")
        Toast.makeText(this@FullImageActivity, "Saved to gallery", Toast.LENGTH_LONG).show()
    }
}
